src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,143 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] internalArray = new Object[10];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size == internalArray.length) {
+            int newLength = (int) (internalArray.length * 1.5);
+            Object[] internalArrayTemp = new Object[newLength];
+            System.arraycopy(internalArray,0, internalArrayTemp, 0, internalArray.length);
+            internalArray = internalArrayTemp;
+        }
+        internalArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist, size of array = "
+                    + size);
+        }
+        if (size == internalArray.length) {
+            int newLength = (int) (internalArray.length * 1.5);
+            Object[] internalArrayTemp = new Object[newLength];
+            System.arraycopy(internalArray,0, internalArrayTemp, 0, internalArray.length);
+            internalArray = internalArrayTemp;
+        }
+        if (size == index) {
+            internalArray[size] = value;
+        } else {
+            Object[] internalArrayTemp = new Object[internalArray.length];
+            System.arraycopy(internalArray,0, internalArrayTemp, 0, index);
+            System.arraycopy(internalArray,index, internalArrayTemp, index + 1, size - index);
+            internalArrayTemp[index] = value;
+            internalArray = internalArrayTemp;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            throw new NullPointerException("List<> shouldn't be null");
+        }
+        if (list.size() + size > internalArray.length) {
+            Object[] internalArrayTemp = new Object[list.size() + size];
+            System.arraycopy(internalArray,0, internalArrayTemp, 0, size);
+            System.arraycopy(toArray(list),0, internalArrayTemp, size, list.size());
+            internalArray = internalArrayTemp;
+            size = internalArray.length;
+            return;
+        }
+        System.arraycopy(toArray(list),0, internalArray, size, list.size());
+        size += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0) {
```

Comment:
[Don’t create repeating code](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#dont-create-repeating-code). Fix it in all places